src/main/java/org/sonarlint/intellij/editor/SonarExternalAnnotator.java

```diff
@@ -77,17 +77,16 @@ public void apply(@NotNull PsiFile file, AnnotationContext annotationResult, @No
       });
 
     var toolWindowService = getService(project, SonarLintToolWindow.class);
-    if (toolWindowService.isSecurityHotspotsTabActive()) {
-      var securityHotspots = issueManager.getSecurityHotspotsForFile(file.getVirtualFile());
-      securityHotspots
-        .forEach(securityHotspot -> {
-          // reject ranges that are no longer valid. It probably means that they were deleted from the file, or the file was deleted
-          var validTextRange = securityHotspot.getValidTextRange();
-          if (validTextRange != null) {
-            addAnnotation(project, securityHotspot, validTextRange, holder);
-          }
-        });
-    }
+    var securityHotspots = issueManager.getSecurityHotspotsForFile(file.getVirtualFile());
+    var displayedSecurityHotspots = toolWindowService.getDisplayedSecurityHotspots(securityHotspots);
```

Comment:
In fact I'm not sure we should annotate security hotspots that come from the Report tab. The report is a snapshot, and it could happen that the code changes after the report has been generated, so in this case we could end up annotating the wrong code. Have you checked this with Marco ? I would be in favor of not annotating hotspots that come from the report tab at all

Reply:
Removed after further discussions :+1: